### ISSUE TITLE
Update documentation explain debug mode

### ISSUE THREAD

**Issue by Salaboy** [wildfly/wildfly-maven-plugin#51]
I've just created this question in the JBoss forums, https://developer.jboss.org/message/934456#934456
It would be nice to update the docs to explain how to use the debug mode with the plugin. 
Is it possible? It is just a simple arg to the standalone.sh command (--debug)

**Comment by blackat** [comment]
Hi,
I have tried to user the following configuration

    <configuration>
        <javaOpts>
            <javaOpt>-agentlib:jdwp=transport=dt_socket,address=8787,server=y,suspend=y</javaOpt>
        </javaOpts>
    </configuration>

but I would like to start the application deployed in debug mode.
For instance with tomcat or jetty plugin, inside intellij, I can do right click on the maven goal and click on debug.

If I do the same with the wildfly-maven-plugin the debugger is does not attach to the server. Is there a way to debug a deployed application inside an IDE with the wildfly maven plugin?

__Workaround.__ For the time being I have a configuration in the IDE starting the goal `mvn widlfy:run`, when the war file has been built it waits on port 8787, then I start another configuration on that port in debug mode, so the deploy process starts.

**Comment by jamezp** [comment]
What version of the plugin are you using?

**Comment by blackat** [comment]
The version is 1.0.0.alpha4

**Comment by jamezp** [comment]
Ah that `<javaOpts/>` wasn't available util version 1.1.0.Alpha1. The current alpha release is 1.1.0.Alpha4. If you want to stick with the 1.0.x (1.0.2.Final being the current final) you'd need to use `<jvmArgs>-agentlib:jdwp=transport=dt_socket,address=8787,server=y,suspend=y</jvmArgs>`.

**Comment by blackat** [comment]
Sorry I have updated the answer...

**Comment by jamezp** [comment]
ha, well that should work then. It's working for me locally on 1.1.0.Alpha4.
```
[INFO] --- wildfly-maven-plugin:1.1.0.Alpha4:run (default-cli) @ simple-servlet ---
[INFO] JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.65-3.b17.fc23.x86_64/jre
[INFO] JBOSS_HOME=/home/jperkins/projects/jboss/examples/javaee/simple-servlet/target/wildfly-run/wildfly-9.0.1.Final

[INFO] Server is starting up. Press CTRL + C to stop the server.
Nov 03, 2015 11:28:11 AM org.xnio.Xnio <clinit>
INFO: XNIO version 3.3.1.Final
Listening for transport dt_socket at address: 5005
Nov 03, 2015 11:28:11 AM org.xnio.nio.NioXnio <clinit>
INFO: XNIO NIO Implementation Version 3.3.1.Final
Nov 03, 2015 11:28:11 AM org.jboss.remoting3.EndpointImpl <clinit>
INFO: JBoss Remoting version 4.0.9.Final
```

I just changed the port to 5005 is all.

Where is the configuration defined? Is it in a profile or an execution?

**Comment by blackat** [comment]
The configuration is defined in an execution, I get your same logging but I guess you run `mvn wildly:run`, then the execution stops waiting the start of another task attaching to that port, then the execution goes on with the deployment of the application, doesn't it?

**Comment by jamezp** [comment]
Ah yes. I'm sorry I misunderstood the original question. I don't know how the IDE's handle the debugging. I just attach with a remote debugger. This would be the same as using the `--debug` option on the standalone startup script.

What IDE are you using? Most of them; eclipse, IDEA and Netbeans, have an option to for the IDE to manage the starting, stopping and debugging for the server. I'd recommend that if you don't want to attach the debugger remotely.

**Comment by cainj13** [comment]
Wow, just saw this.  Would have been much faster than [PR#82] (https://github.com/wildfly/wildfly-maven-plugin/pull/82)!  Guess I should have rtfm'd first.

Is that PR still useful from a usability standpoint?

**Comment by jamezp** [comment]
Moved to https://issues.jboss.org/browse/WFMP-31
